### PR TITLE
Anndata integration

### PR DESCRIPTION
**Pull request by Intron7**
# Changes
* support for anndata
* support for 64 bit floats
* enables more dense array support
* cunnData deprecated
* plotting removed
* switched `neighbors` from tools to preprocessing
* updates docs

### CODE REVIEW

File: docs/Usage_Principles.md [hunk 1]
```diff
@@ -6,36 +6,46 @@
...
-```{image} _static/cunndata.svg
-:width: 500px
-```
+The workflow of *rapids-singlecell* is basically the same as *scanpy's*. The main difference is the speed at which *rsc* can analyze the data. For more information please checkout the notebooks and API documentation.
 
-The {class}`~rapids_singlecell.cunnData.cunnData` object replaces {class}`~anndata.AnnData` for preprocessing. All {mod}`~.pp` and {mod}`~.pl` functions are aimed towards cunnData. {attr}`~.X` and {attr}`~.layers` are stored on the GPU. The other components are stored in the host memory.
+### AnnData setup
 
-## Workflow
+With the release of version 0.10.0 {class}`~anndata.AnnData` support GPU arrays and Sparse Matrices.
```
Comment (Zethson) [inline docs/Usage_Principles.md:15]:
```suggestion
With the release of version 0.10.0 {class}`~anndata.AnnData` supports GPU arrays and Sparse Matrices.
```

File: docs/Usage_Principles.md [hunk 2]
```diff
@@ -6,36 +6,46 @@
...
+Rapids-singlecell leverages this capability to perform analyses directly on GPU-enabled {class}`~anndata.AnnData` objects. This also leads to the depreciation of {class}`~rapids_singlecell.cunnData.cunnData` and it's removal in early 2024.
+
+To get your {class}`~anndata.AnnData` object onto the GPU you can set {attr}`~anndata.AnnData.X` or each {attr}`~anndata.AnnData.layers` to a GPU based matrix.
+
+```
+adata.X = cpx.scipy.sparse.csr_matrix(adata.X)  # moves `.X` to the GPU
+adata.X = adata.X.get() # moves `.X` back to the GPU
+```
+
+You can also use the {mod}`rapids_singlecell.utils` to move arrays and matrices.
```
Comment (Zethson) [inline docs/Usage_Principles.md:26]:
```suggestion
You can also use {mod}`rapids_singlecell.utils` to move arrays and matrices.
```

File: docs/Usage_Principles.md [hunk 3]
```diff
@@ -6,36 +6,46 @@
...
+
+```
+rsc.utils.anndata_to_GPU(adata) # moves `.X` to the GPU
+rsc.utils.anndata_to_CPU(adata) # moves `.X` to the CPU
+```
 
 ### Preprocessing
 
-The preprocessing is handled by {class}`~rapids_singlecell.cunnData.cunnData` and `cunnData_funcs`. The latter is import as {mod}`~.pp` and {mod}`~.pl` to mimic the behavior of scanpy.
+The preprocessing can handled by {class}`~anndata.AnnData` and {class}`~rapids_singlecell.cunnData.cunnData`. It offers accelerated versions of functions within {mod}`scanpy.pp`.
```
Comment (Zethson) [inline docs/Usage_Principles.md:35]:
```suggestion
The preprocessing can  be handled by {class}`~anndata.AnnData` and {class}`~rapids_singlecell.cunnData.cunnData`. It offers accelerated versions of functions within {mod}`scanpy.pp`.
```

File: docs/Usage_Principles.md [hunk 4]
```diff
@@ -55,3 +65,11 @@ rsc.dcg.run_mlm(mat=adata, net=net, source='source', target='target', weight='we
 acts_mlm = dc.get_acts(adata, obsm_key='mlm_estimate')
 sc.pl.umap(acts_mlm, color=['KLF5',"FOXA1", 'CellType'], cmap='coolwarm', vcenter=0)
 ```
+
+### cunnData (depreciated)
```
Comment (Zethson) [inline docs/Usage_Principles.md:69]:
```suggestion
### cunnData (deprecated)
```

File: src/rapids_singlecell/cunnData/__init__.py [hunk 5]
```diff
@@ -615,29 +624,48 @@ def uns_keys(self) -> List[str]:
         """List keys of unstructured annotation."""
         return sorted(self._uns.keys())
 
-    def to_AnnData(self):
+    def to_AnnData(self, device: Literal["cpu", "gpu"] = "cpu") -> AnnData:
         """
         Takes the cunnData object and creates an AnnData object
+        Params
```
Comment (flying-sheep) [inline src/rapids_singlecell/cunnData/__init__.py:630]:
```suggestion

        Params
```

**Comment by LouisFaure** [comment]
Looking forward for this PR! Once done, I will add my two cents by including the `read_mtx` function I proposed previously.

**Pull request merged**